File: src/parser/go/main/utils/issue/issue.go
package issue

import (
	"fmt"
	"os"
	"strconv"
)

func Issue_hint(filename string, line int, col int, message string) {
	itype := "\033[32;1mHint:\033[0m"
	fileinfo := "\033[1m" + filename + "(" + strconv.Itoa(line) + ", " + strconv.Itoa(col) + ")\033[0m"

	fmt.Println(fileinfo + " " + itype + " " + message)
}

func Issue_warn(filename string, line int, col int, message string) {
	itype := "\033[33;1mWarning:\033[0m"
	fileinfo := "\033[1m" + filename + "(" + strconv.Itoa(line) + ", " + strconv.Itoa(col) + ")\033[0m"

	fmt.Println(fileinfo + " " + itype + " " + message)
}

func Issue_error(filename string, line int, col int, message string) {
	itype := "\033[31;1mError:\033[0m"
	fileinfo := "\033[1m" + filename + "(" + strconv.Itoa(line) + ", " + strconv.Itoa(col) + ")\033[0m"

	fmt.Println(fileinfo + " " + itype + " " + message)
	os.Exit(1)
}

File: src/parser/js2/helpers/issue.js
#!/usr/bin/env node

"use strict";

// \033: [https://stackoverflow.com/a/10170631]

function issue_hint(filename, line, col, message) {
	let itype = "\u001b[32;1mHint:\u001b[0m";
	let fileinfo = "\u001b[1m" + filename + "(" + line + ", " + col + ")\u001b[0m";

	console.log(fileinfo + " " + itype + " " + message);
}

function issue_warn(filename, line, col, message) {
	let itype = "\u001b[33;1mWarning:\u001b[0m";
	let fileinfo = "\u001b[1m" + filename + "(" + line + ", " + col + ")\u001b[0m";

	console.log(fileinfo + " " + itype + " " + message);
}

function issue_error(filename, line, col, message) {
	let itype = "\u001b[31;1mError:\u001b[0m";
	let fileinfo = "\u001b[1m" + filename + "(" + line + ", " + col + ")\u001b[0m";

	console.log(fileinfo + " " + itype + " " + message);
	process.exit();
}

module.exports = { issue_hint, issue_warn, issue_error };
